File: ITechQuiz/Models/Survey.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ITechQuiz.Models
{
    public class Survey
    {
        [Key]
        [Required]
        public string Name { get; set; }

        [MaxLength(50)]
        [Required]
        public string Title { get; set; }

        [Required]
        public Type Type { get; set; }

        [MaxLength(150)]
        public string Subtitle { get; set; }

        [DataType(DataType.Date)]
        [Required]
        public DateTime CreatedDate { get; set; }

        public List<Question> Questions { get; set; }

    }

    public enum Type
    {
        ForStatistics,
        ForQuiz
    }
}

File: ITechQuiz/client/src/app/models/survey.ts
import {Question} from "./question";
import {Category} from "./category";

export class Survey {
    public constructor(
        public id?: string,
        public title?: string,
        public type?: string,
        public subtitle?: string,
        public createdDate?: string,
        public updatedDate?: string,
        public lastPassageDate?: string,
        public questions: Question[] = [],
        public isAnonymousAllowed?: boolean,
        public isMultipleAnswersAllowed?: boolean,
        public categories: Category[] = []
    ) {
    }
}